### ISSUE TITLE
[Hold for PR 24041] [$1000] Dev: Web - Composer disappears while switching chats in high traffic account

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22947]
This is held on https://github.com/Expensify/App/pull/24041

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Switch between large chats in a high traffic account

## Expected Result:
Composer remains visible

## Actual Result:
Composer disappears, then later reappears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** Dev 1.3.40-1
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/36ce712b-7efa-48cf-9a27-c63f8d955401

**Expensify/Expensify Issue URL:**
**Issue reported by:** Amy Evans
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689360443816699

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d355d230175a068f</li>
        <li>Upwork Job ID: 1681049473425895424</li>
        <li>Last Price Increase: 2023-07-24</li>
    </ul>
</details>

**Comment by conorpendergrast** [comment]
@amyevans I wasn't able to see this on v1.3.41-2 (Staging), so outside of dev. Have you only found this on dev so far?

**Comment by amyevans** [comment]
I reproduced it on prod, albeit a lot less pronounced:

https://github.com/Expensify/App/assets/7277067/a3459543-258f-43e0-885e-aca0178acc99

Can you try joining the SF lounge room ([SO](https://stackoverflowteams.com/c/expensify/questions/16148/16149#16149)) and see if you experience it after that?

**Comment by conorpendergrast** [comment]
Ok perfect, thanks Amy! I was able to reproduce this **very mildly** on conor+ht@conorpendergrast.com on Production using that Lounge room 👍

**Comment by conorpendergrast** [comment]
I'll default to external for this, but let me know if you think this should be internal

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
While switching chats in high-traffic accounts, the composer disappears for a moment and is displayed at a later time.

### What is the root cause of that problem?
The root cause of the problem is the loading of draft comments using withOnyx which prevents the rendering of the composer for some time.

### What changes do you think we should make in order to solve the problem?
The current behavior shows an empty space within the ReportFooter while the draft comments are being fetched by the composer. This is not a good UI experience.
As the draft comments must be available before rendering the composer, I propose that we move the onyxdata fetch from `ReportActionCompose` to `ReportFooter`. 
This would ensure that the composer is shown only when it is ready with the contents of draft comments.

Please have a look at the test videos that are before and after the changes

Before Fix:

https://github.com/Expensify/App/assets/3004659/5d33bfca-5306-43ba-a913-f82ff27a78bd

After Fix:

https://github.com/Expensify/App/assets/3004659/d4124753-0923-4324-8881-b8293746032f

### What alternative solutions did you explore? (Optional)

I also considered fetching the draft comments when the report screen is getting loaded. But, then, that will delay the overall loading of the report screen and it didn't seem to be a good user experience to me. Since the draft comment is specific to the composer, delaying the load of the footer seems to be a better solution than delaying the load of the report screen.

**Comment by rojiphil** [comment]
## Proposal

[Updated proposal](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543). Added Solution 1 which is more promising to me. @rushatgabhane  Kindly review.

**Comment by rushatgabhane** [comment]
@rojiphil sorry but code diffs aren't accepted as proposals. could you please explain your changes in simple words

**Comment by rojiphil** [comment]
@rushatgabhane  Sorry for the extra details. I am still learning the tricks of writing proposals here. Anyways, I have updated the proposal with only relevant details. Hopefully, this time it is good enough. Kindly review.

**Comment by rojiphil** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543)

**Comment by rojiphil** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543) @rushatgabhane, have further updated the proposal with relevant links to code. Hopefully, this should further help in understanding the proposal. Kindly review. Thanks

**Comment by rushatgabhane** [comment]
@rojiphil i really appreciate the detailed proposal. The root cause is well explained. This bug is caused because the composer is waiting for draft comments to load from onyx.

I like your solution to use get the draft comments only after the composer has been mounted using `onyxSubscribe()`
C+ reviewed 🎀 👀 🎀

**Comment by rushatgabhane** [comment]
TIL: `onyxSubscribe()`

**Comment by rushatgabhane** [comment]
Hi, looks like this issue will be fixed after https://github.com/Expensify/App/issues/22644 is closed.
It solves the root cause of the issue by immediately rendering components.

**Comment by rushatgabhane** [comment]
Okay, we still need to set `initWithStoredValues: false`

Right @rojiphil ?

**Comment by rojiphil** [comment]
> Right @rojiphil ?

Yes, Right.

**Comment by rojiphil** [comment]
[Update from slack discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1690444080646489?thread_ts=1690308156.961499&cid=C01GTK53T8Q): Using `initWithStoredValues: false` is ruled out as the current values from storage are ignored and not sent back for consumption. And, we need it here.

**Comment by conorpendergrast** [comment]
Looks like we're still waiting for updated proposal by @rojiphil, is that correct?

**Comment by rojiphil** [comment]
@rushatgabhane,@aldo-expensify Based on the slack discussion, my thoughts are still to go with the original proposal as the discussion has not conclusively led to another direction. What are your thoughts?

**Comment by aldo-expensify** [comment]
@rojiphil did you set some artificial delay in the code to record the videos in [your proposal](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543)? just asking because it looks like it takes a very very long time to load the draft comments.

**Comment by rojiphil** [comment]
> @rojiphil did you set some artificial delay in the code to record the videos in [your proposal](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543)? just asking because it looks like it takes a very very long time to load the draft comments.

@aldo-expensify There is no artificial delay in the code. I notice this delay when I use my high-traffic account. Hardware and network could also be additional reasons. Usually, I see the delay the first time and it is about 2-3 seconds before composer shows up. However, non high-traffic accounts are quite fast.

**Comment by aldo-expensify** [comment]
I see, thanks, that is a pretty bad user experience in general. I honestly wouldn't care about the composer taking time to appear because the app is extremely laggy overall.
I would work on this trying to improve the performance on the app, but I think that would be a new GH issue.

Thoughts?

**Comment by rojiphil** [comment]
@aldo-expensify I just checked this issue against the latest staging version. This version also has onyx cache improvements. However, as expected, this issue still remains. With the same high-traffic account, the overall experience is very good. I think onyx cache improvements has made the UX experience much better. But, tbh, I think this issue still needs to be addressed. As suggested in [slack here](https://expensify.slack.com/archives/C01GTK53T8Q/p1690444623761339?thread_ts=1690308156.961499&cid=C01GTK53T8Q), we can use onyx connect/disconnect to resolve this problem. Attaching the video of the latest staging version for reference.

https://github.com/Expensify/App/assets/3004659/20073b08-d760-40c5-b9a5-c11c03e8d7bd

**Comment by rushatgabhane** [comment]
i appreciate your consistency to push this issue forward @rojiphil 


@aldo-expensify i think we should go forward with their [proposal](https://github.com/Expensify/App/issues/22947#issuecomment-1643368543).
c+ reviewed 🎀 👀 🎀

**Comment by hannojg** [comment]
@rojiphil I haven't read everything here, just wanted to let you know, that the latest staging version doesn't have the cache improvements yet, as they were temporarily reverted 😅

**Comment by rojiphil** [comment]
> thanks, i think we should wait for the cache improvements to kick in before making a decision then

@rushatgabhane  @aldo-expensify  I am ok with waiting if that is the decision here. But, I am not sure how this helps as per the above experiment with onyx cache PR version and [this slack comment here](https://expensify.slack.com/archives/C01GTK53T8Q/p1690354333018179?thread_ts=1690308156.961499&cid=C01GTK53T8Q).

**Comment by conorpendergrast** [comment]
Ok, looks like this is on hold for https://github.com/Expensify/App/issues/22644 again! I'll mark it as such

**Comment by conorpendergrast** [comment]
Actually looks like we're held on https://github.com/Expensify/App/pull/24041

**Comment by conorpendergrast** [comment]
Still on hold!

**Comment by conorpendergrast** [comment]
Oooh https://github.com/Expensify/App/pull/24041 is now merged! Getting closer

**Comment by conorpendergrast** [comment]
Still held on https://github.com/Expensify/App/pull/24041!

**Comment by aldo-expensify** [comment]
Same ^

**Comment by conorpendergrast** [comment]
Same same

**Comment by conorpendergrast** [comment]
@sophiepintoraetz I'm on parental leave; re-assigning! This has been triaged and we're waiting for https://github.com/Expensify/App/pull/24041 to **deploy**  before we re-test and confirm it's fixed

**Comment by sophiepintoraetz** [comment]
@rushatgabhane @aldo-expensify - are we still holding here?

**Comment by aldo-expensify** [comment]
From looking at https://github.com/Expensify/App/pull/24041, it was merged, but I don't see our typical deploy comments by Melvin-bot so I'm a bit confused.

I think we should just:

1. Remove the HOLD
1. Confirm this problem still exists

**Comment by aldo-expensify** [comment]
I think this is not happening anymore:


https://github.com/Expensify/App/assets/87341702/c78422d9-377d-4f9e-9e82-439dc86d93c2

The composer remains there even if the chat is still loading

@amyevans since you are the reporter, does the video above look like it confirms that this is not longer a problem? I used network tool to make the connection slow

**Comment by amyevans** [comment]
It looks good to me! But I'd also like @shawnborton to confirm since he was the original reporter 😄

**Comment by shawnborton** [comment]
Cool, we can close and reopen if we see it start happening again.
